eitprocessing/filters/butterworth_filters.py

```diff
@@ -0,0 +1,187 @@
+from typing import Literal
+import numpy as np
+import numpy.typing as npt
+from scipy import signal
+from . import TimeDomainFilter
+
+
+MIN_ORDER = 1
+MAX_ORDER = 10
+FILTER_TYPES = ["lowpass", "highpass", "bandpass", "bandstop"]
+
+
+class ButterworthFilter(TimeDomainFilter):
+    """Butterworth filter for filtering in the time domain.
+
+    Generates a low-pass, high-pass, band-pass or band-stop digital Butterworth filter of order
+    `order`.
+
+    ``ButterworthFilter`` is a wrapper of the `scipy.butter()` and `scipy.filtfilt()` functions.
+
+    Args:
+        filter_type: The type of filter to create.
+        cutoff_frequency: Single frequency (lowpass or highpass filter) or tuple containing two
+            frequencies (bandpass and bandstop filters).
+        order: Filter order.
+        sample_frequency: Sample frequency of the data to be filtered.
+        override_order: Whether to raise an exception if the order is larger than the maximum of
+            10. Defaults to False.
+
+    Examples:
+        >>> t = np.arange(0, 100, 0.1)
+        >>> signal = np.sin(t) + 0.1 * np.sin(10 * t)
+        >>> lowpass_filter = ButterworthFilter('lowpass', 45, 4, 250)
+        >>> filtered_signal = lowpass_filter.apply_filter(signal)
+    """
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        filter_type: Literal["lowpass", "highpass", "bandpass", "bandstop"],
+        cutoff_frequency: float | tuple[float, float],
+        order: int,
+        sample_frequency: float,
+        override_order: bool = False,
+    ):
+        if filter_type not in FILTER_TYPES:
+            raise ValueError(
+                "The filter type should be one of "
+                f"{', '.join(FILTER_TYPES)}, not '{filter_type}'."
+            )
+        self.filter_type = filter_type
```

Comment:
I think it would be nice if a parent class `ButterworthFilter` is created by a user, it is automatically assigned to the correct child class.

See e.g. `_set_vendor_class` method in sequency.py:
https://github.com/EIT-ALIVE/eitprocessing/blob/f3cf45e5d6b4ed096dc89fefdd2ae3353d27ddb7/eitprocessing/binreader/sequence.py#L105-L121